pennylane/devices/qubit/simulate.py

```diff
@@ -316,6 +347,334 @@ def simulate_partial(k):
     return measure_final_state(circuit, state, is_state_batched, rng=rng, prng_key=meas_key)
 
 
+# pylint: disable=too-many-arguments, dangerous-default-value
+def simulate_tree_mcm(
+    circuit: qml.tape.QuantumScript,
+    debugger=None,
+    initial_state=None,
+    mcm_active=None,
+    mcm_samples=None,
+    **execution_kwargs,
+) -> Result:
+    """Simulate a single quantum script with native mid-circuit measurements.
+
+    Args:
+        circuit (QuantumTape): The single circuit to simulate
+        rng (Union[None, int, array_like[int], SeedSequence, BitGenerator, Generator]): A
+            seed-like parameter matching that of ``seed`` for ``numpy.random.default_rng``.
+            If no value is provided, a default RNG will be used.
+        prng_key (Optional[jax.random.PRNGKey]): An optional ``jax.random.PRNGKey``. This is
+            the key to the JAX pseudo random number generator. If None, a random key will be
+            generated. Only for simulation using JAX.
+        debugger (_Debugger): The debugger to use
+        interface (str): The machine learning interface to create the initial state with
+        initial_state (TensorLike): Initial state vector
+        mcm_active (dict): Mid-circuit measurement values or all parent circuits of ``circuit``
+        mcm_samples (dict): Mid-circuit measurement samples or all parent circuits of ``circuit``
+
+    Returns:
+        tuple(TensorLike): The results of the simulation
+    """
+    interface = execution_kwargs.get("interface", None)
+
+    samples_present = any(isinstance(mp, SampleMP) for mp in circuit.measurements)
+    postselect_present = any(
+        op.postselect is not None for op in circuit.operations if isinstance(op, MidMeasureMP)
+    )
+    if postselect_present and samples_present and circuit.batch_size is not None:
+        raise ValueError(
+            "Returning qml.sample is not supported when postselecting mid-circuit "
+            "measurements with broadcasting"
+        )
+
+    #########################
+    # shot vector treatment #
+    #########################
+    if circuit.shots.has_partitioned_shots:
+        results = []
+        for s in circuit.shots:
+            aux_circuit = qml.tape.QuantumScript(
+                circuit.operations,
+                circuit.measurements,
+                shots=qml.measurements.Shots(s),
+                trainable_params=circuit.trainable_params,
+            )
+            results.append(simulate_tree_mcm(aux_circuit, debugger, **execution_kwargs))
+        return tuple(results)
+
+    #######################
+    # main implementation #
+    #######################
+
+    def init_dict(d):
+        return {} if d is None else d
+
+    mcm_active = init_dict(mcm_active)
+    mcm_samples = init_dict(mcm_samples)
+
+    circuit_base, circuit_next, op = circuit_up_to_first_mcm(circuit)
+    # we need to make sure the state is the all-wire state
```

Comment:
DQ attempts to use the smallest state possible for a given circuit. So if the first circuit segment only has operations/measurements on wire-0, say in a 20-qubit circuit, it will use a 2-element SV to simulate it. This cause complications transfering a state from segment to segment so I'm forcing the use of the full SV throughout the tree exploration.

Reply:
That makes sense, thanks!